### PR TITLE
fix: various fixes on concepts for v4 and some rephrasing

### PR DESCRIPTION
**Pull request by jeremenichelli**
Initially this PR solves https://github.com/webpack/webpack.js.org/issues/2035 issue.

I also change some parts and code snippets that are not valid with v4 release.

FInally, I encounter a lot of issues with code snippets and content in each section. There were lots of sentences saying exactly the same thing over and over, probably tangling more the understanding an entry user needs on what each concept means in webpack.

Would your words on my changes, the intention was to summarize and simplify the content to be a little more consumable.

You can check the formatted version of the concepts page [here](https://github.com/webpack/webpack.js.org/blob/762a338f28e361d70ae494d1ff12b330441b4a1b/src/content/concepts/index.md) which might help review the content.

### CODE REVIEW

File: src/content/concepts/index.md [hunk 1]
```diff
@@ -10,13 +10,16 @@ contributors:
   - TheDutchCoder
   - adambraimbridge
   - EugeneHlushko
+  - jeremenichelli
 ---
 
-At its core, *webpack* is a _static module bundler_ for modern JavaScript applications. When webpack processes your application, it recursively builds a _dependency graph_ that includes every module your application needs, then packages all of those modules into one or more _bundles_.
+At its core, **webpack** is a _static module bundler_ for modern JavaScript applications. When webpack processes your application, it internally builds a _dependency graph_ which maps every module your project needs, then it generates one or more _bundles_.
```
Comment (EugeneHlushko) [inline src/content/concepts/index.md]: maybe its me but sounds more natural as `project needs and generates one...`

**Comment by EugeneHlushko** [comment]
Overall i found it much more readable, had one small suggestion but lets see what others can say

File: src/content/concepts/index.md (hunk 1 above)
Comment (jeremenichelli) [reply src/content/concepts/index.md]: Agree that sounds better like that. Just pushed the changed.

File: src/content/concepts/index.md [hunk 2]
```diff
@@ -67,20 +68,16 @@ module.exports = {
 
 In the example above, we use the `output.filename` and the `output.path` properties to tell webpack the name of our bundle and where we want it to be emitted to. In case you're wondering about the path module being imported at the top, it is a core [Node.js module](https://nodejs.org/api/modules.html) that gets used to manipulate file paths.
 
-T> You may see the term **emitted** or **emit** used throughout our documentation and [plugin API](/api/plugins). This is a fancy term for 'produced' or 'discharged'.
-
-T> The `output` property has [many more configurable features](/configuration/output) and if you like to know more about the concepts behind the `output` property, you can [read more in the concepts section](/concepts/output).
+T> The `output` property has [many more configurable features](/configuration/output) and if you like to know more about the concepts behind it, you can [read more in the ouput section](/concepts/output).
```
Comment (montogeek) [inline src/content/concepts/index.md]: `ouput` typo

**Pull request merged**